### PR TITLE
[master < ] Enable runtime configurable settings

### LINKED ISSUE #474: Fix Bolt HELLO message response doesn't contains "server" (default value empty)
**Memgraph version**
2.3.1

**Environment**
Windows, Docker, PHP Bolt library

**Describe the bug**
After connecting to memgraph over bolt first message is "hello" to login into. Server response for this message should be map with keys "server" and "connection_id" (optional "hints"). Currently response contains only "connection_id". 

**To Reproduce**
1. Connect to database over bolt and run hello message
2. Analyze response

**Expected behavior**
Response has to contain key "server". Missing key "server" have to contain server agent string identifier like "memgraph/2.3.1".

### PR DESCRIPTION
**Pull request by andrejtonev**
* Add run-time configurable timeout.
* Set the default Bolt server name and allow run-time change.
* Add log-level
* Add also-log-to-stderr

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [ ] Write a release note here, including added/changed clauses
- [ ] Tag someone from docs team in the comments

closes #474

### CODE REVIEW

File: src/flags/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+add_library(mg-flags STATIC audit.cpp
+                            bolt.cpp
+                            general.cpp
+                            isolation_level.cpp
+                            log_level.cpp
+                            memory_limit.cpp
+                            run_time_configurable.cpp)
+target_include_directories(mg-flags PUBLIC ${CMAKE_SOURCE_DIR}/include)
+target_link_libraries(mg-flags PUBLIC spdlog::spdlog mg-settings)
```
Comment (Ignition) [inline src/flags/CMakeLists.txt:9]: Unfortunately not that independent

File: src/flags/general.cpp [hunk 2]
```diff
@@ -11,9 +11,13 @@
 
 #include "general.hpp"
 
+#include "glue/auth_global.hpp"
+#include "storage/v2/config.hpp"
```
Comment (Ignition) [inline src/flags/general.cpp:15]: mg-flags depends on storage

**Review by Ignition — COMMENTED** [review]
Does the move from gflags remove them from `--help`?

File: src/flags/general.cpp [hunk 3]
```diff
@@ -142,7 +146,7 @@ DEFINE_string(pulsar_service_url, "", "Default URL used while connecting to Puls
 // Query flags.
 
 // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
-DEFINE_double(query_execution_timeout_sec, 600,
+DEFINE_double(query_execution_timeout_sec, -1,
               "Maximum allowed query execution time. Queries exceeding this "
               "limit will be aborted. Value of 0 means no limit.");
```
[the remark this answers is not in the document]
Comment (andrejtonev) [reply src/flags/general.cpp:151]: The gflag shouldn't be used directly anymore. The `-1` is just a simple way to determine if the user specified the timeout via the command line. 
If the user specified it, we will use that value. If we read `-1`, we will use the saved value from the storage.
Do you think this makes sense? Or should we reset to the default if the user didn't specify anything (even if there is a saved value in the settings)?

File: src/flags/run_time_configurable.cpp [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+#include "flags/log_level.hpp"
+#include "spdlog/cfg/helpers-inl.h"
+#include "spdlog/spdlog.h"
+#include "utils/exceptions.hpp"
+#include "utils/settings.hpp"
+#include "utils/string.hpp"
+
+namespace {
+// Bolt server name
+constexpr auto kServerNameSettingKey = "server.name";
```
[the remark this answers is not in the document]
Comment (andrejtonev) [reply src/flags/run_time_configurable.cpp:25]: should be `const char*`

File: tests/integration/run_time_settings/runner.py [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+def execute_test(memgraph_binary: str, tester_binary: str, flag_tester_binary: str, executor_binary: str) -> None:
+    storage_directory = tempfile.TemporaryDirectory()
+    memgraph_args = [memgraph_binary, "--data-directory", storage_directory.name]
+
+    print("\033[1;36m~~ Starting run-time settings check test ~~\033[0m")
+
+    print("\033[1;34m~~ server.name and query.timeout ~~\033[0m")
+    # Check default flags
+    run_test(flag_tester_binary, memgraph_args, "Neo4j/v5.11.0 compatible graph database server - memgraph", "600")
```
[the remark this answers is not in the document]
Comment (andrejtonev) [reply tests/integration/run_time_settings/runner.py:167]: @gitbuda I just copy/pasted the name you gave. Is there a reason it's lowercase?

File: src/flags/CMakeLists.txt (hunk 1 above)
Comment (andrejtonev) [reply src/flags/CMakeLists.txt:9]: Made used storage parts be header-only and added mg-utils.

**Comment by andrejtonev** [comment]
@vpavicic 

Added new run-time configurable settings:
1) Server name
2) Query execution timeout
3) Log level
4) Log to stderr

1. Bolt server name can now be changed at run-time by executing the following query
`SET DATABASE SETTING 'server.name' TO 'new-name'`. The updated name persist between session. The user can override the name by updating the settings as above, or by defining the command-line argument as before.
The default server name has also been updated to `Neo4j/v5.11.0 compatible graph database server - Memgraph`.

2. To change the query execution timeout in seconds run `SET DATABASE SETTING 'query.timeout' TO '1234'`.
The updated timeout takes effect on subsequent queries. The query timeout value persists between session and can be updated via the above query or command-line argument.

3. Log level can be set at run-time by calling `SET DATABASE SETTING 'log.level' TO 'TRACE'`.
Allowed values are: TRACE, DEBUG, INFO, WARNING, ERROR, CRITICAL.
The log level is not session persistent and will take on the value of the command-line argument at startup (default: WARNING).

4. User can enable/disable logging to stderr by calling `SET DATABASE SETTING 'log.to_stderr' TO 'true/false'`. The value does not persist between session and will always take on the value of the command-line argument (default: false).

**Review by Ignition — APPROVED** [review]

**Pull request merged**